core/src/main/java/org/mapfish/print/metrics/ApplicationStatus.java

```diff
@@ -28,28 +35,57 @@ class ApplicationStatus extends HealthCheck {
   protected Result check() throws Exception {
     long waitingJobsCount = jobQueue.getWaitingJobsCount();
     if (waitingJobsCount == 0) {
-      return Result.healthy("No print job is waiting in the queue.");
+      return createResult(Result.healthy("No print job is waiting in the queue."));
     }
 
     String health = ". Number of print jobs waiting is " + waitingJobsCount;
 
     if (jobManager.getLastExecutedJobTimestamp() == null) {
-      return Result.unhealthy("No print job was ever processed by this server" + health);
+      return createResult(Result.unhealthy("This server never processed a print job" + health));
     } else if (hasThisServerPrintedRecently()) {
       // WIP (See issue https://github.com/mapfish/mapfish-print/issues/3393)
       if (waitingJobsCount > maxNbrPrintJobQueued) {
-        return Result.unhealthy(
-            "WIP: Number of print jobs queued is above threshold: "
-                + maxNbrPrintJobQueued
-                + health);
+        return createResult(
+            Result.unhealthy(
+                "WIP: Number of print jobs queued is above threshold: "
+                    + maxNbrPrintJobQueued
+                    + health));
       } else {
-        return Result.healthy("This server instance is printing" + health);
+        return createResult(Result.healthy("This server instance is printing" + health));
       }
     } else {
       throw notificationForBrokenServer();
     }
   }
 
+  private Result createResult(final Result current) {
```

Comment:
I don't understand what's this method for? Honestly, it looks like a hack.

Reply:
It creates Unhealthy Result when we have an "unhealthy" counter different from 0

Reply:
As I remember, it's precisely what we want to remove?

This kind of "unhealthy" counter => no pod restart => no unhealthy